source/brailleDisplayDrivers/braillePen.py

```diff
@@ -0,0 +1,182 @@
+#brailleDisplayDrivers/braillePen.py
+#A part of NonVisual Desktop Access (NVDA)
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+#Copyright (C) 2014 Harpo Sp. z o. o. <http://www.harpo.com.pl>
+
+import braille
+import queueHandler
+from logHandler import log
+from ctypes import *
+from ctypes.wintypes import *
+import time
+import config
+import inputCore
+import hwPortUtils
+import brailleInput
+import wx
+import serial
+
+READ_INTERVAL = 50
+BAUD_RATE = 115200
+TIMEOUT = 0.1
+
+BP12_BRAILLE_KEYS = ("dot1", "dot2", "dot3", "dot4", "dot5", "dot6","dot7","dot8" )
+
+
+
+
+def _findPorts():
+	for portInfo in hwPortUtils.listComPorts(onlyAvailable=True):
+		try:
+			btName = portInfo["bluetoothName"]
+		except KeyError:
+			continue
+		if btName.startswith("BraillePen") or btName.startswith("EasyLink"):
+			yield portInfo["port"]
+
+class BrailleDisplayDriver(braille.BrailleDisplayDriver):
+	name = "braillePen"
+	description = _("BraillePen Slim")
+
+	@classmethod
+	def check(cls):
+		return True
+
+	def __init__(self):
+		super(BrailleDisplayDriver,self).__init__()
+		self.numCells = 0
+		self._ser = None
+		
+		for port in _findPorts():
+			try:
+				self._ser = serial.Serial(port, baudrate=BAUD_RATE, timeout=TIMEOUT, writeTimeout=TIMEOUT, parity=serial.PARITY_NONE)
+			except serial.SerialException:
+				log.info("BraillePen open port error")
+				continue
+			
+			break
+		else:
+			raise RuntimeError("BraillePen not found")
+		
+		self._readTimer = wx.PyTimer(self.handleResponses)
+		self._readTimer.Start(READ_INTERVAL)
+
+	def terminate(self):
+		try:
+			self._readTimer.Stop()
+			self._readTimer = None
+			super(BrailleDisplayDriver, self).terminate()
+		finally:
+			self._ser.close()
+		
+
+	def handleResponses(self):
+		try:
+		   while self._ser is not None and self._ser.inWaiting():
+				braille, command = self._readPacket()
```

Comment:
Suspicious: isn't this the same routine used on other drivers such as BrailleNote driver?